server.go

```diff
@@ -487,7 +493,36 @@ func (sp *serverPeer) OnBlock(p *peer.Peer, msg *wire.MsgBlock, buf []byte) {
 // accordingly.  We pass the message down to blockmanager which will call
 // QueueMessage with any appropriate responses.
 func (sp *serverPeer) OnInv(p *peer.Peer, msg *wire.MsgInv) {
-	sp.server.blockManager.QueueInv(msg, sp)
+	if !cfg.BlocksOnly {
+		if len(msg.InvList) > 0 {
+			sp.server.blockManager.QueueInv(msg, sp)
+		}
+		return
+	}
+
+	newInv := wire.NewMsgInv()
```

Comment:
I'd suggest specifying a size hint here in order to cut down on the allocations from growing the backing array.  

```Go
	newInv := wire.NewMsgInvSizeHint(uint(len(msg.InvList)))
```

It does mean the backing array would be larger than it has to be in the case the remote is an older version and txns end up getting filtered out, but that case should be relatively rare.